### DISCUSSION TITLE
8bitDo Zero Compatibility with version 0.5.0

### DISCUSSION THREAD

**Discussion by sushyoshi** [ndeadly/MissionControl#319] [Q&A]
Hello,

I am unable to get my 8bitDo Zero (the first one) to work with mission control. When I try to pair the device it doesnt show that it ahs been paired, however when I go to the mission control app it shows the device has been paired. Pressing any button does not do anything. The 8bitDo Zero controller seems to have a constant blue light but every 5 seconds it blinks again. I believe it cannot establish a stable connection.

 Does anyone else is having this issue? I am running atmosphere version 19.1 on 12.0.0 firmware switch.

Thanks in advance.

V.

**Comment by sushyoshi** [comment]
Also I would like to add that I searched on the issues tab on this github about this device problems and some people seem to get it to work on  mission control version 0.3.0, however when I install that version on my switch atmosphere fails to load.

**Reply by ndeadly** [reply]
Yeah, this is to be expected. Mission control versions are tightly coupled with atmosphere releases. If you use an old version like 0.3.0 you need to use an atmosphere release from around that time too. Obviously an old atmosphere release may not work with your current firmware though.

**Comment by ndeadly** [comment]
I would suggest first updating your firmware and atmosphere versions. 12.0.0 rewrote parts of the bluetooth service and quite a few controllers stopped pairing due to some issues introduced that appear to have been resolved in 12.0.1 and upwards.

If you still have problems after updating, let me know and I'll help troubleshoot.

**Reply by sushyoshi** [reply]
Thank you for your fast reply.

I did as you told me, I have updated the switch firmware to 12.1.0 and atmosphere to 0.19.5.

The problem is still there: I am able to pair the controller on windows mode (start + R), however the controller does not appear on the Change Grip/Order Menu. I know that the controller is paired because when running the homebrew app btdb (Bluetooth Database) it shows the 8BitDo Zero controller as paired., along with the joycons. I tried clearing the data base and redo the pairing process and this does not solve the problem.

The controller LEDs are showing some strange signals: Blue LED stable for 5 seconds and then it rapidly blinks for around 1 second, and returns to stable blue led for 5 seconds. Keeps doing this cycle so I assume it is not connecting correctly.
 I tried to pair the controller on Gamepad mode and same issue, it pairs but does not connect.

Any thoughts what can I try next?

**Reply by ndeadly** [reply]
Can the firmware for this controller be updated? Might be with checking that out too.

If that doesn't help, try installing this debug build that will create a log in `sdmc:/bluetooth-mitm.log` and post it here after trying to pair/connect the controller.

https://github.com/ndeadly/MissionControl/files/6826165/MissionControl-0.5.0-dbg-21c541d.zip

**Reply by sushyoshi** [reply]
Hello. Sorry for the late answer.

I installed that debug build of mission control and tried to pair the controller.  Then I went to the btdb app and cleared the database to start fresh. Then I went to pair the controller and first paired my pro controller successfully. Afterward, I try to pair the 8BitDo Zero controller and was unsuccessful. Here is the log I got:
[bluetooth-mitm.log](https://github.com/ndeadly/MissionControl/files/6872141/bluetooth-mitm.log)

Hope it can help. Thanks.

**Reply by ndeadly** [reply]
No problems. Looking at the log, it appears the controller is pairing successfully and being assigned a handler but then maybe getting disconnected. I wonder if it has anything to do with this SetTsi command I recently changed the handling of...

See if this build makes any difference for you
[MissionControl-0.5.0-8bitdo-zero-tsi-465ffa5.zip](https://github.com/ndeadly/MissionControl/files/6872153/MissionControl-0.5.0-8bitdo-zero-tsi-465ffa5.zip)

**Reply by sushyoshi** [reply]
Thats it. You got it right!. Fully working now. All buttons working. Excellent job! 

Thank you very much.

V.

**Reply by ndeadly** [reply]
Great, I'll add the fix to the next release
